File: 13-RomanToInteger/13-RomanToInteger.py
# Last updated: 2/11/2026, 11:08:45 AM
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """

        values = {'I': 1, 'V':5, 'X':10,'L':50,'C':100,'D':500, 'M':1000}
    
        if len(s) == 1:
            return values[s]

        left = 0

        total = 0

        while (left < len(s)):
            print("left ", s[left])
            if left+1<len(s):
                if values[s[left]] < values[s[left+1]]:
                    total+=values[s[left+1]]-values[s[left]]
                    left+=2
                else:
                    total+=values[s[left]]
                    left+=1
            else:
                total+=values[s[left]]
                left+=1

        return total

        

File: 13-RomanToInteger/13-RomanToInteger.java
// Last updated: 2/11/2026, 11:08:48 AM
class Solution {
    public int romanToInt(String s) {
        HashMap<Character, Integer> table = new HashMap <Character, Integer>();
        table.put('I', 1);
        table.put('V', 5);
        table.put('X', 10);
        table.put('L', 50);
        table.put('C', 100);
        table.put('D', 500);
        table.put('M', 1000);
        int total = 0;
        for (int n = 0 ; n < s.length()-1; n++){
            if(table.get(s.charAt(n)) < table.get(s.charAt(n+1))){
                total -= table.get(s.charAt(n));
            }
            else{
                total += table.get(s.charAt(n));
            }
        }
        total+= table.get(s.charAt(s.length()-1));
        return total;
    }
}